src/ert/analysis/_es_update.py

```diff
@@ -715,17 +689,40 @@ def analysis_IES(
                 f"No active observations for update step: {update_step.name}."
             )
 
-        noise = rng.standard_normal(size=(len(observation_values), S.shape[1]))
-        iterative_ensemble_smoother.fit(
-            S,
-            observation_errors,
-            observation_values,
-            noise=noise,
-            ensemble_mask=ens_mask,
-            inversion=ies.InversionType(module.ies_inversion),
-            truncation=module.enkf_truncation,
-            param_ensemble=param_ensemble,
+        # if the algorithm object is not passed, initialize it
+        if sies_smoother is None:
+            # The sies smoother must be initialized with the full parameter ensemble
+            # Get relevant active realizations
+            param_groups = list(source_fs.experiment.parameter_configuration.keys())
+            parameter_ensemble_active = _all_parameters(
+                source_fs, iens_active_index, param_groups
+            )
+            sies_smoother = ies.SIES(
+                parameters=parameter_ensemble_active,
+                covariance=observation_errors**2,
+                observations=observation_values,
+                seed=rng,
+                inversion=inversion,
+                truncation=analysis_config.enkf_truncation,
+            )
+
+            # Keep track of iterations to calculate step-lengths
+            sies_smoother.iteration = 1
+
+        # Calculate step-lengths to scale SIES iteration
+        step_length = sies_step_length(sies_smoother.iteration)
+
+        # Propose a transition matrix using only active realizations
+        # TO-DO: Verify (by doing the math) math of slicing W after realizations is okay
```

Comment:
Can this TODO be an issue instead?

Reply:
Yes, it should be an issue. Made an issue in `iterative_ensemble_smoother` https://github.com/equinor/iterative_ensemble_smoother/issues/196